modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/listk.kt

```diff
@@ -234,13 +234,21 @@ interface ListKMonadCombine : MonadCombine<ForListK> {
   override fun <A> Kind<ForListK, A>.orElse(b: Kind<ForListK, A>): ListK<A> =
     fix() + b.fix()
 
-  override fun <A : Any> Kind<ForListK, A>.some(): ListK<SequenceK<A>> =
+  override fun <A> Kind<ForListK, A>.some(): ListK<SequenceK<A>> =
     if (this.fix().isEmpty()) ListK.empty()
-    else map { generateSequence { it }.k() }.k()
+    else map { Sequence { object : Iterator<A> {
```

Comment:
Isn't this equivalent to `sequenceOf(it)`?

Reply:
Nope. `sequenceOf(it)` generates a sequence of only one element. This way we are generating an infinite sequence of one repeated element.